api/src/main/java/org/pmiops/workbench/api/DisksController.java

```diff
@@ -108,9 +113,14 @@ private List<Disk> findTheMostRecentActiveDisks(List<Disk> disksToValidate) {
     // Iterate original list first to check if disks are valid. Print log if disks maybe in
     // incorrect state to help future debugging.
     // Disk maybe in incorrect state if having additional active state disks.
+    String pdNamePrefix = userProvider.get().getUserPDNamePrefix();
+
     List<Disk> activeDisks =
         disksToValidate.stream()
-            .filter(d -> ACTIVE_DISK_STATUS.contains(d.getStatus()))
+            .filter(
+                d ->
+                    ACTIVE_DISK_STATUS.contains(d.getStatus())
+                        && d.getName().startsWith(pdNamePrefix))
```

Comment:
could we use `Disk.creator` here instead?

Reply:
ah good point! I did see creator is propagated correctly!